### ISSUE TITLE
Filename shortening

### ISSUE THREAD

**Issue by lassik** [arthurgleckler/validate-html#3]
The compilation buffer says `Output from W3C HTML Validator on "/full/absolute/filename.html"`; the filename can get quite long. Could it be shortened so the compilation buffer's `default-directory` is set to `/full/absolute` and the buffer only says `Output from W3C HTML Validator on "filename.html"`?

Here's the code from PR #1:

```emacs lisp
(let ((file-dir (expand-file-name
                 (if (buffer-file-name)
                     (file-name-directory (buffer-file-name))
                     "~")))
      (filename (and (buffer-file-name)
                     (file-name-nondirectory (buffer-file-name)))))
  ...
  (with-current-buffer output-buffer
    (setq default-directory file-dir)
    (insert "Output from W3C HTML Validator"
            (if filename (format " on %S" filename) "")
            "\n\n")
    ...))
```

**Comment by lassik** [comment]
Since you now require a non-null `(buffer-file-name)` the code can be simplified further.

**Comment by lassik** [comment]
Implemented in commit 394d80f
